app/models/concerns/template_tax.rb

```diff
@@ -0,0 +1,12 @@
+module TemplateTax
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    def find_without_name_collision(name)
+      from_local = find_or_initialize_by :name => name
+      from_global = unscoped.find_or_initialize_by :name => name
+      from_local.errors.add(:name, _("cannot be used, please choose another")) if from_local.new_record? && from_global.persisted?
+      from_local
+    end
+  end
+end
```

Comment:
This is bad name for this concern, it should not mention template but rather name and also "Tax" is misleading, perhaps something like TaxonomyCollisionFinder. You can extract `:name` as parameter so it's generic and reusable across our app.